ldhooks/hook_metadata.go

```diff
@@ -0,0 +1,33 @@
+package ldhooks
+
+// HookMetadata contains information about a specific hook implementation.
+type HookMetadata struct {
```

Comment:
Suggestion: `HookMetadata` -> `Metadata`, so you have `ldhooks.Metadata` & `ldhooks.NewMetadata`